Guide/src/reference/architecture/openhcl_servicing.md

```diff
@@ -0,0 +1,173 @@
+
+### Saved State Format
+
+The servicing state uses Protocol Buffers for serialization. For details on save state implementation, compatibility requirements, and best practices, see the [Save State guide](../../dev_guide/contrib/save-state.md).
+
+## Configuration
+
+Servicing behavior is controlled by several environment variables and configuration parameters. These parameters are set by the boot shim and passed to the kernel, init, and OpenHCL processes.
+
+### NVMe Keepalive Configuration
+
+The `OPENHCL_NVME_KEEP_ALIVE` environment variable controls NVMe keepalive behavior. For details on the values and their meanings, see the [KeepAliveConfig rustdocs](https://openvmm.dev/rustdoc/linux/underhill_core/options/enum.KeepAliveConfig.html).
+
+The boot shim (see `openhcl_boot`) infers the configuration based on the detected environment unless explicitly overridden.
+
+### DMA Pool Configuration
+
+The `OPENHCL_IGVM_VTL2_GPA_POOL_CONFIG` parameter controls the VTL2 GPA pool configuration used for the private pool. For details on the parameter values and behavior, see the [Vtl2GpaPoolConfig rustdocs](https://openvmm.dev/rustdoc/linux/openhcl_boot/struct.Vtl2GpaPoolConfig.html).
+
+The boot shim determines pool sizes using heuristics defined in `openhcl_boot` based on the system configuration, unless explicitly overridden.
+
+### Test Scenarios
+
+For testing servicing behavior, the `OPENHCL_TEST_CONFIG` environment variable can be set to simulate specific conditions:
+
+- `SERVICING_SAVE_STUCK`: Causes save operation to wait indefinitely
+- `SERVICING_SAVE_FAIL`: Forces save operation to fail
+- `SERVICING_RESTORE_STUCK`: Causes restore to wait indefinitely
+
+These test scenarios help validate timeout handling and failure recovery. They are specified as values to the `OPENHCL_TEST_CONFIG` environment variable.
+
+## Error Handling
+
+Servicing operations include comprehensive error handling:
+
+### Save Failures
+- Component failures during save are logged with detailed error information
+- Critical state that cannot be saved may prevent servicing from proceeding
+- The host provides a deadline for the save operation, but the timeout is not enforced within VTL2
+
+### Restore Failures
+- Critical component failures prevent VM startup with detailed error messages
+- The correlation ID helps trace issues across the servicing operation
+
+### Timeout Handling
+- The host specifies a timeout deadline when initiating the servicing request
+- OpenHCL attempts to complete the save operation before the deadline
+- If the deadline is exceeded, the host may force termination (timeout enforcement occurs at the host level, not within VTL2)
+
+## Implementation Details
+
+### Memory Management
+
+During servicing:
+- Private pool pages remain allocated and mapped across the servicing operation
+- Shared pool is recreated on demand during the new boot flow (since a new kernel boots, the shared pool allocation is reconstructed as needed)
```

Comment:
Is this 100% true? I think I led you this way, but really I was thinking about the locked memory pool.